pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,395 @@
+        @property
+        @abstractproperty
+    in python 2
+        @abstractproperty
+    """
+
+    if PY2:  # pylint: disable=no-else-return
+        return abstractproperty(func)
+    else:
+        return property(abstractmethod(func))
+
+
+@add_metaclass(DocInheritMeta(style="google", abstract_base_class=True))
+class BaseObjectStore(object):
+    """
+    An abstract class that defines the APIs for the methods of an object store
+    for CIM objects including CIMClass, CIMInstance, and CIMQualifierDeclaration
+    objectsthat constitute a WBEM server repository.  This
+    class provides the abstract methods for creating, accessing, and deleting,
+    CIM objects of a single CIM object type in the repository.
+    """
+
+    def __init__(self, cim_object_type):
+        """
+        Parameters:
+
+            cim_object_type(:term:`CIM object`):
+              The Pywbem cim object type as defined in cim_types.py for the
+              objects in the data store. Used to verify values on create.
+        """
+        self._cim_object_type = cim_object_type
+
+    @abstractmethod
+    def exists(self, name):
+        """
+        Test if CIM object defined by name exists in the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+        Returns:
+
+            True if the object exists; False if it does not exist
+        """
+        pass
+
+    @abstractmethod
+    def get(self, name, copy=True):
+        """
+        Return the CIM object defined by name if it exists in the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store
+
+          copy: If True, returns a copy of the object to insure that modifying
+          the returned object does not change the data store. The default
+          behavior is True .  If copy is False, the object in the object store
+          is returned but if it is modified by the user, the object in the
+          store may be modified also.
+
+        Returns:
+
+            Returns the cim_object identified by the name parameter
+
+        Raises:
+
+            KeyError: cim_object with name not in this object store
+        """
+        pass
+
+    @abstractmethod
+    def create(self, name, cim_object):
+        """
+        Adds the cim_object identified by name to the object store.
+
+        Parameters:
+
+          name(:term:`string` or :class:`~pywbem.CIMInstanceName`):
+            Name by which the object is identified in the object store .
+
+          cim_object(:term:`CIM object` to be added to the object store):
+            The CIM object that will be inserted into the object store. The
+            object is copied into the object store so the user can safely
+            modify the original object without affecting the store.
```

Comment:
Change the type in parenthesis to be just the type, i.e.:
```
cim_object (:term:`CIM object`):
```

Reply:
DONE.

Reply:
Not done - This change did not make it into the last commit.

Reply:
DONE in new commit.